File: src/javascript/syntax.js
import { memberExpression } from "@babel/types";


//define attributes / variables
const functions = {

    addNum: (a,b) => {
        return a+b;
    },

    // string
    myString: () => {
        let c="apple";
        return c;
    },
    
    myBoolean: (a,b) => {
        return a <= b;
    },

    myArray: (array) => {
        return array;
    },

    // //dictionary / objects
    
    member1: (fstName, lstName, ag) => {
        let person = {Firstname:fstName, Lastname:lstName, age:ag};
        return person.Firstname;
    },


    // sample if / else
    
    biggerOut: (a,b) => {
        if (a > b){
            return a;
        }else{
            return b;
        }
    },

// undefined

    myunderfined: (a, b) => {
        a + b;
    },

// functions
    // parameters
    // returns

    myFun: (a, b) => {
        return a * b;
    },

// arrays
    // add to the front
addFront: (addArray) => {
    addArray.unshift(2);
    return addArray;
},

    // add to the end
addEnd: (addArray) => {
    addArray.push(2);
    return addArray;
},
    // update values

addUpdate: (addArray) => {
    addArray[1] = 2;
    return addArray;
},

// loops 
    // for
loopFor: (sw) => {
    for (var i = 0; i < sw; i++);
        return i;
},
    
    // for/in
loopForIn: (myTeam) => {
    var z;
    for (z in myTeam) {
        return z
    }

},
    // while
loopWhile: (sw) => {
    var j = 0
    while (j < sw ){
        j=j+1;
    } return j;
},

    // do while
loopDoWhile: (mySw) => {
    var ourArray = [];
    var k =0
    do{
        ourArray.push(k);
        k++;
    } while (k <5);
    return ourArray

    },

    // forEach (with array and function)
    myForEach: (numArray) => {
        const number = numArray;
        const double = [];
        const newArray = number.forEach((num) => {
            double.push(num*2);
        })
            return double;
    },

    // Objects / Dictionaries
        // declare object
        // lookup key to retrieve value
        
        member1: (fstName, lstName, ag) => {
            let person = {Firstname:fstName, Lastname:lstName, age:ag};
            return person.Firstname;
        }
    


    }

export default functions;

File: src/python/comm200/syntax.py
    # define attributes / variables
    #     number
def give_nums(a):
    nums = a
    return nums
    
    #     string
def give_strs(word):
    strs = word
    return strs

    #     boolean
def myboolean(a, b):
    if a>b:
        return True
    return False

    #     list
def give_list(array):
    the_list = array
    return the_list

    #     dictionary / objects
def give_dictionary(object):
    the_dictionary = object
    return the_dictionary

    #     None
def give_none(z):
    the_none = z
    return the_none

    # sample if / else
def if_else(a, b):
    a= 5
    b= 6
    if a>b:
        return "a is bigger"
    else:
        return "b is bigger"


    # functions
    #     parameters
    #     returns

def add(x, y):
   return x+y

    # lists
    #     add to the front

def add_to_the_front(item, mylist):
    mylist.insert(0, item)
    return mylist
    

    #     add to the end
def add_to_the_end(item, mylist):
    mylist.insert(len(mylist),5)
    return mylist

    #     update values
def update_values(subject, changes, mylist):
    for idx,item in enumerate(mylist):
        if subject in item:
            mylist[idx] = changes
            return mylist

    
    # loops 
    #     for/in
def for_loop(mylist):
    total = 0
    for i in mylist:
        total += i
    return total

    
    #     while:
def while_loop(x):
    j=0
    mylist =[]
    while j < x:
        mylist.append(j)
        j += 1
    return mylist
    
    # Objects / Dictionaries
    #     lookup key to retrieve value

def lookup(fstName, lstName, ag):
    person = {"Firstname":fstName, "Lastname":lstName, "age":ag}
    return person["Firstname"]
    



